sklearn/model_selection/tests/test_validation.py

```diff
@@ -1637,8 +1637,22 @@ def test_fit_and_score_failing():
                        "partition for these parameters will be set to %f. "
                        "Details: \n%s" % (fit_and_score_kwargs['error_score'],
                                           error_message))
+
+    def test_warn_no_trace(msg):
+        split = msg.splitlines()  # note: handles more than '\n'
+        mtb = split[0] + '\n' + split[-1]
+        return warning_message in mtb
     # check if the same warning is triggered
-    assert_warns_message(FitFailedWarning, warning_message, _fit_and_score,
+    assert_warns_message(FitFailedWarning, test_warn_no_trace, _fit_and_score,
```

Comment:
Is this assertion needed? The only difference between `test_warn_no_trace` and `test_warn_trace` is the assertion regarding the traceback.